synapse/cores/storage.py

```diff
@@ -722,6 +723,67 @@ def genStoreRows(self, **kwargs):
         for rows in self._genStoreRows(**kwargs):
             yield rows
 
+    def _loadUpdateProperty(self, mesg):
+        oldprop = mesg[1].get('oldprop')
+        newprop = mesg[1].get('newprop')
+        self._updateProperty(oldprop, newprop)
+        self._updateTagForm(oldprop, newprop)
+
+    def updateProperty(self, oldprop, newprop):
+        '''
+        Do a wholesale replacement of one property with another property.
+
+        Args:
+            oldprop (str): The orginal property which is removed.
+            newprop (str): The property that is updated in place.
+
+        Examples:
+            Rename "inet:tcp4:port" to "inet:tcp4:foobar"::
+
+                nrows = store.updateProperty('inet:tcp4:port', 'inet:tcp4:foobar')
+
+        Notes:
+            This API does  fire syn:core:store:updateprop:pre and syn:core:store:updateprop:post events with the old
+            and new property names in it, before and after the property update is done. This API is primarily designed
+            for assisting with Cortex data migrations.
+
+        Returns:
+            int: Number of rows updated in place.
+        '''
+        if oldprop == newprop:
+            raise s_common.BadPropName(mesg='OldProp and newprop cannot be the same.',
+                                       oldprop=oldprop, newprop=newprop)
+
+        if not isinstance(newprop, str):
+            raise s_common.BadPropName(mesg='newprop must be a str', newprop=newprop)
+
+        self.savebus.fire('core:save:updateproperty', oldprop=oldprop, newprop=newprop)
+        self.fire('syn:core:store:updateprop:pre', oldprop=oldprop, newprop=newprop)
+        nrows = self._updateProperty(oldprop, newprop)
+        self._updateTagForm(oldprop, newprop)
+        self.fire('syn:core:store:updateprop:post', oldprop=oldprop, newprop=newprop, nrows=nrows)
+        return nrows
+
+    def _updateTagForm(self, oldform, newform):
+        adds, dels, darks = [], [], []
+        for i, p, v, t in self.getRowsByProp('syn:tagform:form', oldform):
+            adds.append((i, p, newform, t),)
+            dels.append((i, p, v),)
+
+            for _, _, tag, _ in self.getRowsByIdProp(i, 'syn:tagform:tag'):
+                oldark = '_:*%s#%s' % (oldform, tag)
+                newdark = '_:*%s#%s' % (newform, tag)
+                darks.append((oldark, newdark),)
+
+        if adds:
+            self._addRows(adds)
+
+        for i, p, v in dels:
```

Comment:
We do have a delRowsByProp which accepts a valu, making it a delrowsbypropvalu function.